### ISSUE TITLE
Second factor OTP: Invalid secret is not handled and causes error 500

### ISSUE THREAD

**Issue by phorward** [viur-framework/viur-core#559]
The OTP Secret (otpkey) can be anything currently, and errors are not handled. This causes an error 500 within the login process.
```
Traceback (most recent call last):
  File "/layers/google.python.pip/pip/lib/python3.10/site-packages/viur/core/request.py", line 249, in processRequest
    self.findAndCall(path)
  File "/layers/google.python.pip/pip/lib/python3.10/site-packages/viur/core/request.py", line 577, in findAndCall
    res = caller(*newArgs, **newKwargs)
  File "/layers/google.python.pip/pip/lib/python3.10/site-packages/viur/core/modules/user.py", line 568, in otp
    validTokens = self.generateOtps(token["otpkey"], token["otptimedrift"])
  File "/layers/google.python.pip/pip/lib/python3.10/site-packages/viur/core/modules/user.py", line 543, in generateOtps
    currHash = hmac.new(bytes.fromhex(secret), asBytes(slot), hashlib.sha1).digest()
ValueError: non-hexadecimal number found in fromhex() arg at position 0
```

**Comment by sveneberth** [comment]
Is this solved by https://github.com/viur-framework/viur-core/pull/802?

**Comment by phorward** [comment]
> Is this solved by #802?

yes!
